hail/src/test/scala/is/hail/utils/RowIntervalSuite.scala

```diff
@@ -1,28 +1,55 @@
+
+class RowIntervalSuite extends HailSuite {
+  lazy val t = TStruct("a" -> TInt32, "b" -> TInt32, "c" -> TInt32)
+  lazy val pord = PartitionBoundOrdering(t)
+
+  def assertContains(i: Interval, point: Row, shouldContain: Boolean = true): Unit = {
+    val c = i.contains(pord, point)
+    if (shouldContain)
+      assert(c)
+    else
+      assert(!c)
+
+    val tt = t.truncate(point.length)
+    val irRepIntervalType = RVDPartitioner.intervalIRRepresentation(tt)
+    val intervalIRRep = RVDPartitioner.intervalToIRRepresentation(i.coarsen(tt.size), tt.size)
+
+    val in1 = In(0, irRepIntervalType)
+    val in2 = In(1, tt)
+
+    assertEvalsTo(
+      ir.invoke("partitionIntervalContains", TBoolean, in1, in2),
+      args = FastIndexedSeq((intervalIRRep, irRepIntervalType), (point, tt)),
+      shouldContain)(ExecStrategy.compileOnly)
+  }
 
   @Test def testContains() {
-    assert(Interval(Row(0, 1, 5), Row(1, 2, 4), true, true).contains(pord, Row(1, 1, 3)))
-    assert(Interval(Row(0, 1, 5), Row(1, 2, 4), true, true).contains(pord, Row(0, 1, 5)))
-    assert(!Interval(Row(0, 1, 5), Row(1, 2, 4), false, true).contains(pord, Row(0, 1, 5)))
-    assert(!Interval(Row(0, 1, 5), Row(1, 2, 4), true, false).contains(pord, Row(1, 2, 4)))
-
-    assert(Interval(Row(0, 1), Row(1, 2, 4), true, true).contains(pord, Row(0, 1, 5)))
-    assert(!Interval(Row(0, 1), Row(1, 2, 4), false, true).contains(pord, Row(0, 1, 5)))
-    assert(Interval(Row(0, 1), Row(0, 1, 4), true, true).contains(pord, Row(0, 1, 4)))
-    assert(!Interval(Row(0, 1), Row(0, 1, 4), true, false).contains(pord, Row(0, 1, 4)))
-
-    assert(Interval(Row(0, 1), Row(2, 1, 4), true, false).contains(pord, Row(1)))
-    assert(!Interval(Row(0, 1), Row(2, 1, 4), true, false).contains(pord, Row(0)))
-    assert(!Interval(Row(0, 1), Row(2, 1, 4), true, false).contains(pord, Row(2)))
+    assertContains(Interval(Row(0, 1, 5), Row(1, 2, 4), true, true), Row(1, 1, 3))
+    assertContains(Interval(Row(0, 1, 5), Row(1, 2, 4), true, true), Row(0, 1, 5))
+    assertContains(Interval(Row(0, 1, 5), Row(1, 2, 4), false, true), Row(0, 1, 5), shouldContain = false)
+    assertContains(Interval(Row(0, 1, 5), Row(1, 2, 4), true, false), Row(1, 2, 4), shouldContain = false)
+
+    assertContains(Interval(Row(0, 1), Row(1, 2, 4), true, true), Row(0, 1, 5))
+    assertContains(Interval(Row(0, 1), Row(1, 2, 4), false, true), Row(0, 1, 5), shouldContain = false)
+    assertContains(Interval(Row(0, 1), Row(0, 1, 4), true, true), Row(0, 1, 4))
+    assertContains(Interval(Row(0, 1), Row(0, 1, 4), true, false), Row(0, 1, 4), shouldContain = false)
+
+    assertContains(Interval(Row(0, 1), Row(1, 2, 4), true, true), Row(0, 1, 5))
+    assertContains(Interval(Row(0, 1), Row(1, 2, 4), false, true), Row(0, 1, 5), shouldContain = false)
+    assertContains(Interval(Row(0, 1), Row(0, 1, 4), true, true), Row(0, 1, 4))
+    assertContains(Interval(Row(0, 1), Row(0, 1, 4), true, false), Row(0, 1, 4), shouldContain = false)
```

Comment:
I'd like to have some tests cover zero-length endpoints too. Maybe
```
assertContains(Interval(Row(), Row(1, 2, 4), true, true), Row(1, 2, 4))
assertContains(Interval(Row(), Row(1, 2, 4), true, false), Row(1, 2, 4), shouldContain = false)
assertContains(Interval(Row(1, 2, 4), Row(), true, true), Row(1, 2, 4))
assertContains(Interval(Row(1, 2, 4), Row(), false, true), Row(1, 2, 4), shouldContain = false)
```

Reply:
oh, good call